.circleci/config.yml

```diff
@@ -20,4 +20,40 @@ jobs:
               git branch --set-upstream-to=origin/${CIRCLE_BRANCH} ${CIRCLE_BRANCH}
               echo $DOCKER_PASS | docker login -u $DOCKER_USER --password-stdin
               ./release.sh
-            fi
\ No newline at end of file
+            fi
+ build_mac:
+  environment:    
+    SOFTWARE: /tmp/circleci-artifacts/software_versions.json
+  macos:
+    xcode: 10.1.0
+  steps:
+    - checkout
+    - restore_cache:
+        key: go-mod-v1-macos-{{ checksum "go.sum" }}
+        paths:
+            - /go/pkg/mod
+    - run: brew install go
```

Comment:
Does/could the go install mess with `/go/pkg/mod`? Should this happen before restoring the cache?